lib/magma/server/update.rb

```diff
@@ -55,4 +57,36 @@ def load_revisions
     log(m.complaints)
     @errors.concat(m.complaints)
   end
+
+  def update_any_file_links
+    # Here, if there are any File attributes in the revisions and
+    #   the storage is configured to use Metis instead of AWS,
+    #   we'll send a request to the Metis "copy" route, using the
+    #   Etna::Client.
+    if Magma.instance.config(:storage).fetch(:provider).downcase == 'metis'
+
+      # Etna::Client.new throws an exception if the host config includes protocol,
+      # i.e. https://metis-dev.etna-development.org will throw an exception
+      raise Etna::BadRequest, 'Storage host should not include protocol' if
+        Magma.instance.config(:storage).fetch(:host).start_with? 'http'
+
+      # Do we get the token from config?
+      client = Etna::Client.new(
+        "https://#{Magma.instance.config(:storage).fetch(:host)}",
+        Magma.instance.config(:token) ? Magma.instance.config(:token) : 'token')
+
```

Comment:
The token (which is the user's JWT token) should come from the request, you can see in https://github.com/mountetna/etna/blob/c3240af177b26f50fa228304b7e955b7e09f4c43/lib/etna/auth.rb#L68 how to grab it. When we pass this on to metis, we are making the request to Metis as whatever user made the request to magma.